pkg/controllers/sandbox/sandbox_controller.go

```diff
@@ -0,0 +1,198 @@
+	"github.com/vertica/vertica-kubernetes/pkg/vk8s"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+const (
+	verticaDBNameKey = "verticaDBName"
+)
+
+// SandboxConfigMapReconciler reconciles a ConfigMap for sandboxing
+type SandboxConfigMapReconciler struct {
+	client.Client
+	Log         logr.Logger
+	Scheme      *runtime.Scheme
+	Cfg         *rest.Config
+	EVRec       record.EventRecorder
+	Concurrency int
+}
+
+// +kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch
+// +kubebuilder:rbac:groups=vertica.com,resources=verticadbs,verbs=get;list;watch
+
+func (r *SandboxConfigMapReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(
+			&corev1.ConfigMap{},
+			builder.WithPredicates(r.predicateFuncs(), predicate.ResourceVersionChangedPredicate{}),
+		).
+		WithOptions(controller.Options{MaxConcurrentReconciles: r.Concurrency}).
+		Complete(r)
+}
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the ConfigMap object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.1/pkg/reconcile
+func (r *SandboxConfigMapReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	// Generate a unique uuid for each reconcile iteration so we can easily
+	// trace actions within a reconcile.
+	reconcileUUID := uuid.New()
+	log := r.Log.WithValues("configMap", req.NamespacedName, "reconcile-uuid", reconcileUUID)
+	log.Info("starting reconcile of sandbox configmap")
+
+	// Fetch the ConfigMap instance
+	configMap := &corev1.ConfigMap{}
+	err := r.Get(ctx, req.NamespacedName, configMap)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// ConfigMap not found, return and don't requeue
+			log.Info("ConfigMap not found.  Ignoring since object must be deleted")
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "failed to get ConfigMap")
+		return ctrl.Result{}, err
+	}
+
+	// Fetch the VDB found in the configmap
+	vdb := &v1.VerticaDB{}
+	nm := names.GenNamespacedName(configMap, configMap.Data[verticaDBNameKey])
+	var res ctrl.Result
+	if res, err = vk8s.FetchVDB(ctx, r, configMap, nm, vdb); verrors.IsReconcileAborted(res, err) {
+		return res, err
+	}
+	log.WithName(vdb.Name)
```

Comment:
You need to save the log that `WithName` generates. And, actually, I think `WithValues` is better as it will make it clear what the name is referring too.
```suggestion
	log = log.WithValues("verticadb", vdb.Name)
```